src/main/webapp/app/shared/data-table/data-table.component.ts

```diff
@@ -257,7 +256,7 @@ export class DataTableComponent implements OnInit, OnChanges {
         const searchPredicate = (entity: BaseEntity) => {
             return !this.searchEntityFilterEnabled || this.filterEntityByTextSearch(this.entityCriteria.textSearch, entity, this.searchFields);
         };
-        const filteredEntities = compose(filter(searchPredicate), filter(this.customFilter))(this.allEntities);
+        const filteredEntities = this.allEntities.filter(this.customFilter).filter(searchPredicate);
```

Comment:
```suggestion
        const filteredEntities = this.allEntities.filter((entity) => this.customFilter(entity) && searchPredicate(entity));
```
This should produce the same result in a single traversal without allocating an intermediate array.